mixins/cross-compile.mixin

```diff
@@ -0,0 +1,31 @@
+{
+    "build": {
+        "armhf": {
+            "cmake-args": [
+                "-DCMAKE_TOOLCHAIN_FILE=/home/ubuntu/aws-ros-dev/cross_compile/mixins/cmake-toolchains/armhf-linux.cmake"
+            ],
+            "cmake-clean-cache": true,
+            "cmake-clean-first": true,
+            "cmake-force-configure": true,
+            "merge-install": true
+        },
+        "pi_armhf-raspbian_stretch": {
+            "cmake-args": [
+                "-DCMAKE_SYSTEM_NAME=Linux",
```

Comment:
Why do we pass values here which are statically set in the cmake toolchain file?

Reply:
We want to replace the toolchain files with colcon mixins for all supported architectures. Synced with Anas and we will do that in the next iteration.